### PR TITLE
fix(Login): Add redirectUrl to Google login url

### LINKED ISSUE #797: Send redirectUrl to Google login
### Description

When the user tries to access a secured path and logs in with Google, the path is not remembered. We need to send this redirectUrl to Google so that the user is taken to the path once they log in.

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Changes
- When google login button is clicked in the homepage, it will append the redirect url to the google login url

## Test
- Test with PR https://github.com/WISE-Community/WISE-API/pull/176

Closes #797

### CODE REVIEW

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
Works well. I have a code clean up suggestion.

File: src/app/login/login-home/login-home.component.ts [hunk 1]
```diff
@@ -113,13 +113,19 @@ export class LoginHomeComponent implements OnInit {
...
-      redirectUrl = `${redirectUrl}?accessCode=${this.accessCode}`;
+      redirectUrl = this.appendAccessCodeParameter(redirectUrl);
     }
     return redirectUrl;
   }
+
+  private appendAccessCodeParameter(url: string): string {
+    return url.includes('?')
+      ? `${url}&accessCode=${this.accessCode}`
+      : `${url}?accessCode=${this.accessCode}`;
```
Comment (geoffreykwan) [inline src/app/login/login-home/login-home.component.ts]: Should this be rewritten to reduce duplication? Something like
```
return `${url}${url.includes('?') ? '%' : '?'}accessCode=${this.accessCode}`;
```
or
```
const delimiter = url.includes('?') ? '%' : '?';
return `${url}${delimiter}accessCode=${this.accessCode}`;
```

**Review by breity — APPROVED** [review]
Looks good.

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Pull request merged**